src/vs/workbench/services/textfile/common/textFileEditorModel.ts

```diff
@@ -290,11 +290,14 @@ export class TextFileEditorModel extends BaseTextEditorModel implements ITextFil
 				diag('load() - created text editor model', this.resource, new Date());
 
 				return this.backupFileService.loadBackupResource(this.resource).then(backupResource => {
-					let resolveBackupPromise: TPromise<IRawText>;
+					let resolveBackupPromise: TPromise<string | IRawText>;
 
 					// Try get restore content, if there is an issue fallback silently to the original file's content
 					if (backupResource) {
-						resolveBackupPromise = this.textFileService.resolveTextContent(backupResource, BACKUP_FILE_RESOLVE_OPTIONS).then(backup => backup.value, error => content.value);
+						resolveBackupPromise = this.textFileService.resolveTextContent(backupResource, BACKUP_FILE_RESOLVE_OPTIONS).then(backup => {
```

Comment:
@Tyriar this is bad, what if we resolve the text content in another place? everyone has to remove the first line? why not move the entire thing into the backup service and thus have the logic in one place where it should be. We can even get rid of `loadBackupResource` and just have 1 method to call to resolve the backup content (which can be null if there is no backup).